### ISSUE TITLE
git integration

### ISSUE THREAD

**Issue by kianenigma** [sam0x17/docify#14]
end goal: 

```
#[doc = docify::embed!(git: "github.com/paritytech/polkadot-sdk.git", path: "substrate/frame/examples/basic/src/lib.rs", item: my_cool_item)]
```

**Comment by sam0x17** [comment]
another thing I'm contemplating here is being able to specify at the crate level a fallback git url to use for relative paths, so at the beginning of `lib.rs` you could specify something like:

```rust
docify::set_git_fallback!(git: "github.com/paritytech/polkadot-sdk.git", base_path: "path/to/current/crate");
```

And then any embeds that fail, it will use that git repo and path as the fallback and will try to find things from within there. So this way you could still use your relative paths and they will work normally 99% of the time, and then in that weird scenario where they don't work because of `cargo publish` shenanigans, it will use this fallback.

But the basic git feature is also something I want to do anyway because there are some scenarios where that is the only way to do something

**Comment by kianenigma** [comment]
IF and when you would use git, it would be cool if you could enable in option that would also put a permalink to the line of code as a comment in the first line of the snippet. So you would see the code, then you could also click on this link and see it in the original context.

**Comment by sam0x17** [comment]
yeah was thinking about having something like a badges.io button to the permalink or something but yeah this is also in my mental scaffold for this

**Comment by sam0x17** [comment]
one key thing when implementing this: need to make sure to use some sort of crate that provides git cloning, since it is likely that the `git` cli is not available on the crates.io runners

**Comment by kianenigma** [comment]
> one key thing when implementing this: need to make sure to use some sort of crate that provides git cloning, since it is likely that the `git` cli is not available on the crates.io runners

How about just raw GET requests to raw.githubuserconent.com and so on and so forth?

**Comment by sam0x17** [comment]
> > one key thing when implementing this: need to make sure to use some sort of crate that provides git cloning, since it is likely that the `git` cli is not available on the crates.io runners
> 
> How about just raw GET requests to raw.githubuserconent.com and so on and so forth?

could, but would be better to operate at the git level i.e. for people who use non-github git hosting

**Comment by sam0x17** [comment]
yeah looking into it

**Comment by sam0x17** [comment]
support added in 0.3.0, see #29
